src/test/java/spoon/support/reflect/CtModifierHandlerTest.java

```diff
@@ -0,0 +1,67 @@
+package spoon.support.reflect;
+
+import org.junit.Before;
+import org.junit.Test;
+import spoon.Launcher;
+import spoon.reflect.declaration.ModifierKind;
+import spoon.reflect.factory.Factory;
+
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+public class CtModifierHandlerTest {
+
+    LinkedHashSet<ModifierKind> inputModifierKinds = new LinkedHashSet<>(Arrays.asList(
```

Comment:
Why not use Lists here if the order is important? That would also ensure the test *actually fails* if the order is wrong! Currently it just accepts any order as the equals method doesn't take order into account (see the comment further down).

```suggestion
    List<ModifierKind> inputModifierKinds = Arrays.asList(
```

Reply:
I could use lists, but the problem with this is that I need to convert the input data into a set at some point so I can feed it into the setModifiers and setExtendedModifiers methods, which only accept sets. But maybe I could add overloads for these methods that accept lists

Reply:
If the API only takes in a Set I would probably rather order them while printing. Otherwise you need to change all callsites and  users might accidentally break it:
Some types lack `addModifier` methods so users need to fetch them, add them to their own set and re-set them. I wouldn't expect this to make printing non-deterministic. I will raise this concern in the attached issue though.